site/search/Autocomplete.tsx

```diff
@@ -74,17 +80,40 @@ const onSelect: AutocompleteSource<BaseItem>["onSelect"] = ({
 const getItemUrl: AutocompleteSource<BaseItem>["getItemUrl"] = ({ item }) =>
     item.slug as string
 
-// The slugs we index to Algolia don't include the /grapher/ or /explorers/ directories
+// The slugs we index to Algolia don't include grapher/, explorers/, or data-insights/ subdirectories
 // Prepend them with this function when we need them
 const prependSubdirectoryToAlgoliaItemUrl = (item: BaseItem): string => {
     const indexName = parseIndexName(item.__autocomplete_indexName as string)
-    const subdirectory = indexNameToSubdirectoryMap[indexName]
-    switch (indexName) {
-        case SearchIndexName.ExplorerViews:
-            return `${subdirectory}/${item.explorerSlug}${item.viewQueryParams}`
-        default:
-            return `${subdirectory}/${item.slug}`
-    }
+    return match(indexName)
+        .with(SearchIndexName.ExplorerViews, () => {
+            return urljoin(
+                EXPLORERS_ROUTE_FOLDER,
+                item.explorerSlug as string,
+                item.viewQueryParams as string
+            )
+        })
+        .with(SearchIndexName.Charts, () => {
+            return urljoin(BAKED_GRAPHER_URL, item.slug as string)
+        })
+        .with(SearchIndexName.Pages, () => {
+            if (
+                item.type === WordpressPageType.Country ||
+                item.type === WordpressPageType.Other
+            ) {
+                return `/${item.slug}`
+            }
+            return getCanonicalUrl(BAKED_BASE_URL, {
+                slug: item.slug as string,
+                content: {
+                    type: item.type as OwidGdocType,
+                },
+            })
+        })
+        .with(
+            SearchIndexName.ExplorerViewsMdimViewsAndCharts,
+            () => `${BAKED_GRAPHER_URL}/${item.slug}`
+        )
```

Comment:
this won't fire for now since we're not sourcing this index, but when we do we'll have a wrong path prefix ("grapher") for explorer views.